arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowFieldType.java

```diff
@@ -55,9 +45,20 @@ enum ArrowFieldType {
     this.clazz = clazz;
   }
 
+  ArrowFieldType(Class<?> clazz, int precision, int scale) {
```

Comment:
I think the other constructors should call this one.
There are some values in the type system for "unspecified", maybe these could be reused.
Using 0 may not be always safe.